src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -49,6 +60,57 @@ internal bool AutomaticRedirection
                 _automaticRedirection = value;
             }
         }
+        internal bool UseProxy
+        {
+            get
+            {
+                if (_proxyPolicy == CurlProxyUsePolicy.UseDefaultProxy)
+                {
+                    return false;
+                }
+                return _proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy;
```

Comment:
Yes.  That piece of code for returning the value of 'UseProxy' is fine:
`return (_proxyPolicy != CurlProxyUsePolicy.DoNotUseProxy);`

This conversation got confusing because @stephentoub  was saying that perhaps this code was wrong based on the other code above.  It turns out that this code is correct.  But the code above it is still wrong.